ddtrace/_logger.py

```diff
@@ -0,0 +1,70 @@
+import logging
+from logging.handlers import RotatingFileHandler
+import os
+
+from ddtrace.internal.utils.formats import asbool
+
+
+DEFAULT_FILE_SIZE_BYTES = 15 << 20  # 15 MB
+
+
+def configure_ddtrace_logger():
+    # type: () -> None
+    """Configures ddtrace log levels and file paths.
+
+    Customization is possible with the environment variables:
+        ``DD_TRACE_DEBUG``, ``DD_TRACE_LOG_FILE_LEVEL``, and ``DD_TRACE_LOG_FILE``
+
+    By default, when none of the settings have been changed, ddtrace loggers
+        inherit from the root logger in the logging module and no logs are written to a file.
+
+    When DD_TRACE_DEBUG has been enabled:
+        - Propagate logs up so logs show up in the application logs if a file path wasn't provided
+        - Logs are routed to a file when DD_TRACE_LOG_FILE is specified, using the log level in DD_TRACE_LOG_FILE_LEVEL.
+        - Child loggers inherit from the parent ddtrace logger
+
+    Note(s):
+        1) The ddtrace-run logs under commands/ddtrace_run do not follow DD_TRACE_LOG_FILE if DD_TRACE_DEBUG is enabled.
+            This is because ddtrace-run calls ``logging.basicConfig()`` when DD_TRACE_DEBUG is enabled, so
+            this configuration is not applied.
+        2) Python 2: If the application is using DD_TRACE_DEBUG=true, logging will need to be configured,
+            ie: ``logging.basicConfig()``.
+
+    """
+    ddtrace_logger = logging.getLogger("ddtrace")
+
+    configure_ddtrace_debug_logger(ddtrace_logger)
+    configure_ddtrace_file_logger(ddtrace_logger)
+
+
+def configure_ddtrace_debug_logger(logger):
```

Comment:
can we make this and the other one private?

```suggestion
def _configure_ddtrace_debug_logger(logger):
```

Reply:
otherwise this new structure lgtm (@Kyle-Verhoog agrees)

Reply:
I'll rename it to `_configure_ddtrace_debug_logger` and `_configure_ddtrace_file_logger`. Marking as resolved.